### PR TITLE
fix: show zero value metrics [DET-4686]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

- show zero value metrics (currently treated as empty)
- fix handling of negative numbers (treated as decimals always currently)
- avoid showing integers as decimal or scientific notations

Before:
Screen Shot 2021-02-04 at 10 46 15 AM

After:
Screen Shot 2021-02-04 at 10 45 56 AM

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/pages/TrialDetails.tsx [hunk 1]
```diff
@@ -179,7 +179,7 @@ const TrialDetailsComp: React.FC = () => {
     const metricRenderer = (metricName: MetricName) => {
       const metricCol = (_: string, record: Step) => {
         const value = extractMetricValue(record, metricName);
-        return value ? <HumanReadableFloat num={value} /> : undefined;
+        return value != null ? <HumanReadableFloat num={value} /> : undefined;
```
Comment (apizzini) [inline webui/react/src/pages/TrialDetails.tsx:182]: `!=` instead of `!==` on purpose?

Comment (hkang1) [reply webui/react/src/pages/TrialDetails.tsx:182]: Ah yes, so `!==` will check against only `null`. `value != null` is the same as `value !== null && value !== undefined`. `value == null` is equivalent to `value === null || value === undefined`.

Basically it protects against both `null` and `undefined`.

Comment (hkang1) [reply webui/react/src/pages/TrialDetails.tsx:182]: https://stackoverflow.com/questions/16607761/why-null-undefined-is-true-in-javascript

Comment (apizzini) [reply webui/react/src/pages/TrialDetails.tsx:182]: Thanks :)

**Review by apizzini — APPROVED** [review]
LGTM

**Pull request merged**